### PR TITLE
Testing

### PR DESCRIPTION
**Pull request by IanMitchell**
Work in Progress for #10. Call out any additional tests you think I'm missing!

### CODE REVIEW

**Comment by IanMitchell** [comment]
Hm, apparently Mocha outputs to `console.log`. The log output from ARX-7 gets intermingled with the test output during tests; Not sure there's a way to 'mute' the main application without muting the test output too. Might be worth looking into a plugin like debug for non-test output.

**Comment by IanMitchell** [comment]
We have better logging now, but it's a little inconsistent in places. I'll probably give it a pass-over later on

**Comment by IanMitchell** [comment]
Hm... the async in the YouTube plugin is causing problems. The test doesn't really wait; it's synchronous in nature. I can change the Client `say` method to use an async callback, but the problem is there's not a clear way of tying into that, since the YouTube plugin wouldn't interface with it. I might be able to rig something up where I set it up before I call the YouTube plugin, but I'm not quite sure how to do that yet. The next little bit to look into, I guess.

File: test/commands/test_choose.js [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+import {Choose} from "./../../src/commands/choose";
+
+let client = new Client();
+let choose = new Choose(client);
+
+describe('Choose', () => {
+  describe('General Usage', () => {
+    it('should respond in correct channel', () => {
+      choose.message('Mocha', '#test', '.c this, that', null);
+      assert.equal('#test', client.lastTarget);
```
Comment (kunaldes) [inline test/commands/test_choose.js]: Should these be assert.strictEqual?

File: src/arx7.js [hunk 2]
```diff
@@ -1,13 +1,15 @@
+import debug from 'debug';
 import irc from 'irc';
-import config from './config';
+import config from './../config';
```
Comment (kunaldes) [inline src/arx7.js]: (nit) ./../ seems odd to me. Why not just ../?

**Comment by kunaldes** [comment]
Left a few notes but overall LGTM. Not the biggest fan of putting unimplemented tests as failures, but since we don't have continuous build it doesn't really matter.

**Comment by IanMitchell** [comment]
Yeah, and it's more temporary - I won't merge until they're all finished. Setting up continuous testing would be interesting... hm. Made the changes you suggested, will push with the next batch of tests

**Comment by IanMitchell** [comment]
Getting closer. 
```
  36 passing (621ms)
  20 pending
  2 failing
```

That doesn't include testing Range though...

**Comment by IanMitchell** [comment]
#20 is fixed in this branch

File: src/commands/choose.js [hunk 3]
```diff
@@ -1,18 +1,26 @@
...
+      let choose = text.match(regex);
 
-    if (choose) {
-      let result = this.choose(choose[1]);
-      this.send(to, `${from}: ${result}`);
-    }
+      if (choose) {
+        let result = this.choose(choose[1]);
+        this.send(to, `${from}: ${result}`);
+        resolve();
```
Comment (Kyhz) [inline src/commands/choose.js]: Do you need this `resolve()`? It'll do it on the next line anyway, right?

Comment (IanMitchell) [inline src/commands/choose.js]: Yup! Nice catch, thanks

**Comment by kunaldes** [comment]
lol

**Pull request closed without merging**